connectors/sources/servicenow.py

```diff
@@ -0,0 +1,346 @@
+                ) as response:
+                    if (
+                        not RUNNING_FTEST
+                        and response.headers.get("Connection") == "close"
+                    ):
+                        raise Exception("Couldn't connect to ServiceNow instance")
+
+                    if is_attachment:
+                        yield response
+                        break
+
+                    fetched_response = await response.read()
+                    if fetched_response == b"":
+                        raise InvalidResponse(
+                            "Request hasn't processed from ServiceNow server"
+                        )
+                    elif not response.headers["Content-Type"].startswith(
+                        "application/json"
+                    ):
+                        raise InvalidResponse(
+                            "Request retrieved with invalid content type to process further"
+                        )
+
+                    json_response = json.loads(fetched_response)
+                    result = json_response["result"]
+
+                    response_length = len(result)
+                    if response_length == 0:
+                        break
+
+                    offset += response_length
+                    retry = 1
+                    yield result
+
+            except Exception as exception:
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}."
+                )
+                await self._sleeps.sleep(RETRY_INTERVAL**retry)
+
+                if isinstance(
+                    exception,
+                    ServerDisconnectedError,
+                ):
+                    await self.session.close()  # pyright: ignore
+
+                if retry == self.retry_count:
+                    raise exception
+                retry += 1
+
+    async def filter_services(self):
+        """Filter services based on service mappings.
+
+        Returns:
+            list, list: Valid service names, Invalid services.
+        """
+
+        try:
+            logger.debug("Filtering services")
+            service_names, invalid_services = [], self.services.copy()
+
+            payload = {"sysparm_fields": "label, name"}
+            async for response in self._api_call(
+                url=ENDPOINTS["TABLE"].format(table="sys_db_object"), params=payload
+            ):
+                for mapping in response:  # pyright: ignore
+                    if mapping["label"] in invalid_services:
+                        service_names.append(mapping["name"])
+                        invalid_services.remove(mapping["label"])
+
+            return service_names, invalid_services
+
+        except Exception as exception:
+            logger.exception(f"Error while filtering services. Exception: {exception}.")
+            raise
+
+    async def ping(self):
+        """Ping to ServiceNow instance."""
```

Comment:
```suggestion
```

IMO we should remove doc strings, when they don't add any additional information, which isn't already present in the method name.